src/main/flight/mixer.c

```diff
@@ -766,51 +743,79 @@ float applyThrottleLimit(float throttle) {
     return throttle;
 }
 
+void mixWithThrottleLegacy(float *motorMix, float motorMixMin, float motorMixMax) {
+    float motorMixDelta = controllerMixRange / 2.0f;
+    float zeroThrottleAuthority = isAirmodeActive() ? 1.0f : 0.5f;
+    float normFactor = controllerMixRange > 1.0f && hardwareMotorType != MOTOR_BRUSHED ? controllerMixRange : 1.0f;
+
+    if (thrustLinearizationLevel && !currentPidProfile->use_throttle_linearization) {
+        // counter compensating thrust linearization on throttle
+        throttle = MOTOR_TO_THRUST(throttle, thrustLinearizationLevel);
+    }
+
+    if (mixerImpl == MIXER_IMPL_LEGACY) {
+        // legacy clipping adjustment
+        if (normFactor > 1.0f) {
+            for (int i = 0; i < motorCount; i++) {
+                motorMix[i] /= normFactor;
+            }
+            // Get the maximum correction by setting offset to center when airmode enabled
+            if (isAirmodeActive()) {
+                throttle = 0.5f;
+            }
+        } else {
+            if (isAirmodeActive() || throttle >
+                                     0.5f) {  // Only automatically adjust throttle when airmode enabled. Airmode logic is always active on high throttle
+                throttle = constrainf(throttle, -motorMixMin, 1.0f - motorMixMax);
+            }
+        }
+    }
+
+    float vbatCompFactor = calculateBatteryCompensationFactor();
+
+    for (int i = 0; i < motorCount; i++) {
+        if (mixerImpl == MIXER_IMPL_SMOOTH) {
+            motorMix[i] += motorMixDelta - motorMixMax; // let's center values around the zero
```

Comment:
Is this needed? In my opinion this line isn't doing anything. motorMixDelta - motorMixMax will always result to 0.

Reply:
You are right, it could also cause problems in those mixers that couple e.g. one 1.0 component with two -0.5 components (tricopters or something like that). I removed this line in the new version of the mixer!
Thank you.